### PR TITLE
IA-3289 Use workbench-libs for Azure calls

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3289

* Moved Azure api calls to https://github.com/broadinstitute/workbench-libs/pull/1005. Added logging to azure calls as part of this. The rest should be exactly the same as before

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
```
org.broadinstitute.dsde.workbench.leonardo.runtimes.RuntimePatchSpec.Patch endpoint should perform a stop/start transition for GCE VM with PD
```

Seen this a few times in hourly run too...

jenkins retest

File: http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changelog.xml [hunk 1]
```diff
@@ -86,6 +86,6 @@
     <include file="changesets/20220118_update_runtime_controlled_resource.xml" relativeToChangelogFile="true" />
     <include file="changesets/20220211_cluster_workspace_unique.xml" relativeToChangelogFile="true" />
     <include file="changesets/20220428_add_service_path.xml" relativeToChangelogFile="true" />
-    <include file="changesets/20220628_add_non_null_for_cloud_context.xml" relativeToChangelogFile="true" />
     <include file="changesets/20220626_add_disk_source.xml" relativeToChangelogFile="true" />
+    <include file="changesets/20220628_add_non_null_for_cloud_context.xml" relativeToChangelogFile="true" />
```
Comment (jdcanas) [inline http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changelog.xml]: looks like this may have been a merge conflict?

Comment (Qi77Qi) [reply http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changelog.xml]: oh dang...good catch 👍 !

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpDockerDAOSpec.scala [hunk 2]
```diff
@@ -160,8 +160,9 @@ class HttpDockerDAOSpec extends AnyFlatSpec with Matchers with BeforeAndAfterAll
         |}
         |""".stripMargin
     val expectedResult =
-      DecodingFailure("Attempt to decode value on failed cursor", List(DownField("config")))
-    decode[ContainerConfigResponse](jsonString) shouldBe Left(expectedResult)
+      DecodingFailure("Missing required field", List(DownField("config")))
+    decode[ContainerConfigResponse](jsonString).swap.toOption.get.getMessage shouldBe expectedResult.leftSideValue
```
Comment (jdcanas) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpDockerDAOSpec.scala:164]: hmm.. why did this PR require updating the DockerDaoSpec?

Comment (Qi77Qi) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpDockerDAOSpec.scala:164]: I think circe version gets updated transitively...so the error message is a bit different in newer version

**Review by jdcanas — APPROVED** [review]

**Pull request merged**